src/gen_rules.ml

```diff
@@ -998,6 +1064,7 @@ module Gen(P : Params) = struct
         ~preprocess:lib.buildable.preprocess
         ~virtual_deps:lib.virtual_deps
     in
+
```

Comment:
One way to automate the generation of `.merlin` files is to write:

```ocaml
let real_requires =
   requires ~dir ~dep_kind ...
in
let requires =
  match Path.extract_build_context dir with
  | Some ("default", remaindir) ->
    Build.path (Path.relative ~dir:remaindir ".merlin") >>> real_requires
  | _ -> real_requires
in
...
```

and return `real_requires` at the end of this function.